src/main/java/de/gematik/pki/gemlibpki/validators/CertificateValidator.java

```diff
@@ -0,0 +1,30 @@
+package de.gematik.pki.gemlibpki.validators;
+
+import de.gematik.pki.gemlibpki.exception.GemPkiException;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+
+import java.security.cert.X509Certificate;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+
+public interface CertificateValidator {
+
+    default void validateCertificate(@NonNull X509Certificate x509EeCert) throws GemPkiException {
+        validateCertificate(x509EeCert, ZonedDateTime.now(ZoneOffset.UTC), new ValidationContext());
+    }
+
+    default void validateCertificate(@NonNull X509Certificate x509EeCert, @NonNull ZonedDateTime referenceDate) throws GemPkiException {
+        validateCertificate(x509EeCert, referenceDate, new ValidationContext());
+    }
+
+    void validateCertificate(@NonNull X509Certificate x509EeCert, @NonNull ZonedDateTime referenceDate, @NonNull ValidationContext context)
+            throws GemPkiException;
+
+    @Getter
+    @Setter
+    class ValidationContext {
+        String ocspAddress;
+    }
```

Comment:
The class is unused atm. Is it really necessary? Furthermore we do not see any tests that explain and show the functionality. We would remove it in a following commit.

Reply:
Indeed the ValidationContext is not really needed and could be removed.

It was introduced to provide a kind of callback mechanism for collecting and providing details about the validation.